mixer/adapter/rbac/rbacStore.go

```diff
@@ -304,8 +424,45 @@ func valueMatch(a interface{}, b string) bool {
 	return false
 }
 
-// Check if all properties defined for the subject are satisfied by the properties from the request.
-func checkSubject(properties map[string]interface{}, subject map[string]string) bool {
+// checkSubjectsMatch returns true if rule subjects and instance subjects are matched. Otherwise
+// returns false.
+func checkSubjectsMatch(subject *rbacproto.Subject, instance *authorization.Instance) bool {
+	userMatch := false
+	groupsMatch := false
+	propertyMatch := false
+
+	if instance.Subject.User != "" {
+		if subject.GetUser() == "*" || subject.GetUser() == instance.Subject.User {
+			userMatch = true
+		}
+	}
+
+	if instance.Subject.Groups != "" {
+		if subject.GetGroup() == "*" || subject.GetGroup() == instance.Subject.Groups {
+			groupsMatch = true
+		}
+	}
+
+	if len(subject.GetProperties()) != 0 {
+		propertyMatch = checkSubjectProperties(instance.Subject.Properties, subject.GetProperties())
+	}
+
+	if instance.Subject.User != "" && instance.Subject.Groups != "" && len(instance.Subject.Properties) != 0 {
```

Comment:
Can we return false early in above if statements and then only return true in the end of this function?

Reply:
Done

Reply:
I meant to remove those if-statements in starting from line 457, see my new comment above.